src/util/DeviceFingerprint.js

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Copyright (c) 2015-2016, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/*jshint latedef:false */
+/*global JSON */
+
+define(['vendor/lib/q', 'jquery'], function (Q, $) {
+
+  var iframeId = 'okta_fingerprint_iframe';
```

Comment:
Since we're creating the iframe ourselves, let's not use an id to identify it. General idea here is that this code is being embedded in other people's pages that we have no control over, so the less things we can put in here that have a chance of conflicts, the better.